src/json.cpp

```diff
@@ -0,0 +1,114 @@
+#include <migraphx/serialize.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/literal.hpp>
+#include <nlohmann/json.hpp>
+#include <migraphx/json.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+using json = nlohmann::json;
+
+void value_to_json(const value& val, json& j);
+
+template <class T>
+void value_to_json(const T& x, json& j)
+{
+    j = x;
+}
+
+void value_to_json(const std::vector<value>& x, json& j)
+{
+    for(auto& v : x)
+    {
+        if(v.get_key().empty())
+        {
+            json jj;
+            value_to_json(v, jj);
+            j.push_back(jj);
+        }
+        // in the case of an object
+        else
+        {
+            value_to_json(v, j);
```

Comment:
I think this can be written as `j[v.get_key()] = v.without_key()`.

Reply:
fixed